src/components/LabelPicker/__tests__/LabelPicker.spec.tsx

```diff
@@ -0,0 +1,340 @@
+        });
+        it('should call onChange with with full label object (e.g. include key-values not included in the type)', async () => {
+            const mockOnChange = jest.fn();
+            rerenderLabelPicker({
+                onChange: mockOnChange,
+                labels: labelsMock,
+            });
+            await userEvent.click(screen.getByRole('button', { name: 'Click me' }));
+
+            await userEvent.click(screen.getAllByRole('checkbox')[3]);
+            expect(mockOnChange).toHaveBeenCalledWith(labelsMock[3], expect.any(Object));
+        });
+        it('should call onAdd when add button is clicked', async () => {
+            const mockOnAdd = jest.fn();
+            rerenderLabelPicker({
+                onAdd: mockOnAdd,
+                labels: labelsMock,
+            });
+            await userEvent.click(screen.getByRole('button', { name: 'Click me' }));
+            await userEvent.type(screen.getByRole('textbox'), 'test');
+            await userEvent.click(screen.getAllByRole('button')[1]);
+
+            expect(mockOnAdd).toHaveBeenCalledTimes(1);
+            expect(mockOnAdd).toHaveBeenCalledWith('test');
+        });
+        it('should call onAdd when add ENTER is pressed', async () => {
+            const mockOnAdd = jest.fn();
+            rerenderLabelPicker({
+                onAdd: mockOnAdd,
+                labels: labelsMock,
+            });
+            await userEvent.click(screen.getByRole('button', { name: 'Click me' }));
+            await userEvent.type(screen.getByRole('textbox'), 'test');
+            await userEvent.keyboard('[Enter]');
+
+            expect(mockOnAdd).toHaveBeenCalledTimes(1);
+            expect(mockOnAdd).toHaveBeenCalledWith('test');
+        });
+        it('should call onClose when dialog is closed due to Done button click', async () => {
+            const mockOnClose = jest.fn();
+            rerenderLabelPicker({
+                onClose: mockOnClose,
+                labels: labelsMock,
+            });
+            await userEvent.click(screen.getByRole('button', { name: 'Click me' }));
+            await userEvent.click(screen.getAllByRole('button')[2]);
+
+            expect(mockOnClose).toHaveBeenCalledTimes(1);
+        });
+        it('should call onClose when dialog is closed due to outer click', async () => {
+            const clickDocument = useDocumentEvent('click');
+            const mockOnClose = jest.fn();
+            rerenderLabelPicker({
+                onClose: mockOnClose,
+                labels: labelsMock,
+            });
+
+            expect(screen.queryAllByRole('dialog').length).toBe(0);
+
+            // open dialog
+            await userEvent.click(screen.getByRole('button', { name: 'Click me' }));
+
+            expect(screen.queryAllByRole('dialog').length).toBeGreaterThan(0);
+
+            // click outside of component
+            act(() => {
+                clickDocument('click');
```

Comment:
This was initially done to be used with Enzyme. Maybe there is a different approach to this using RTL.

Reply:
@andrentk Do you mean to replace  clickDocument('click') or just remove act ?

Reply:
I meant for clickDocument. I think act is already part of userEvent, not 100% on the later.

Reply:
:+1: